src/chrome/chromeDebugAdapter.ts

```diff
@@ -1148,10 +1221,26 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
             // after refreshing the page. This is the only way to allow hitting breakpoints in code that runs immediately when
             // the page loads.
             const script = this.getScriptByUrl(url);
-            const urlRegex = utils.pathToRegex(url, this._caseSensitivePaths);
-            responsePs = breakpoints.map(({ line, column = 0, condition }, i) => {
-                return this.addOneBreakpointByUrl(script && script.scriptId, urlRegex, line, column, condition);
-            });
+
+            // If script has been parsed, script object won't be undefined and we would have the mapping file on the disk and we can directly set breakpoint using that
+            if (this._breakOnLoadStrategy === 'none' || script) {
+                const urlRegex = utils.pathToRegex(url, this._caseSensitivePaths);
+                responsePs = breakpoints.map(({ line, column = 0, condition }, i) => {
+                    return this.addOneBreakpointByUrl(script && script.scriptId, urlRegex, line, column, condition);
+                });
+            } else { // Else if script hasn't been parsed and break on load is active, we need to do extra processing
+
+                // If the strategy is set to regex, we try to match the file where user put the breakpoint through a regex and tell Chrome to put a stop on entry breakpoint there
+                if (this._breakOnLoadStrategy === 'regex') {
```

Comment:
Can you move these few lines into the helper? Then it can check the strategy and do the right thing.